cms/djangoapps/contentstore/views/tests/test_course_index.py

```diff
@@ -330,3 +336,129 @@ def _assert_settings_link_present(response):
 
         self.assertEqual(_get_release_date(response), get_default_time_display(self.course.start))
         _assert_settings_link_present(response)
+
+
+class TestCourseReIndex(CourseTestCase):
+    """
+    Unit tests for the course outline.
+    """
+
+    TEST_INDEX_FILENAME = "test_root/index_file.dat"
+
+    def setUp(self):
+        """
+        Set up the for the course outline tests.
+        """
+
+        super(TestCourseReIndex, self).setUp()
+
+        self.course.start = datetime.datetime(2014, 1, 1, tzinfo=pytz.utc)
+        modulestore().update_item(self.course, self.user.id)
+
+        self.chapter = ItemFactory.create(
+            parent_location=self.course.location, category='chapter', display_name="Week 1"
+        )
+        self.sequential = ItemFactory.create(
+            parent_location=self.chapter.location, category='sequential', display_name="Lesson 1"
+        )
+        self.vertical = ItemFactory.create(
+            parent_location=self.sequential.location, category='vertical', display_name='Subsection 1'
+        )
+        self.video = ItemFactory.create(
+            parent_location=self.vertical.location, category="video", display_name="My Video"
+        )
+
+        self.html = ItemFactory.create(
+            parent_location=self.vertical.location, category="html", display_name="My HTML",
+            data="<div>This is my unique HTML content</div>",
+
+        )
+
+        # create test file in which index for this test will live
+        with open(self.TEST_INDEX_FILENAME, "w+") as index_file:
```

Comment:
JSON doesn't have a native serialization format for DateTime objects, but everyone just uses [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format. I believe Django actually has a custom JSON encoder built in that can handle DateTime objects for you. See the [Django documentation about serialization](https://docs.djangoproject.com/en/1.4/topics/serialization/), or just import it yourself:
```python
import json
from django.core.serializers.json import DjangoJSONEncoder
result = json.dumps(obj, cls=DjangoJSONEncoder)
```
See the [json module docs](https://docs.python.org/2/library/json.html#json.dump).